pkg/flare/sysprobe/archive_linux.go

```diff
@@ -3,36 +3,94 @@
+
+	sysprobeclient "github.com/DataDog/datadog-agent/cmd/system-probe/api/client"
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+)
 
 // GetSystemProbeConntrackCached queries conntrack/cached, which uses our conntracker implementation (typically ebpf)
 // to return the list of NAT'd connections
-func GetSystemProbeConntrackCached(socketPath string) ([]byte, error) {
-	probeUtil, err := net.GetRemoteSystemProbeUtil(socketPath)
+func GetSystemProbeConntrackCached(client *http.Client) ([]byte, error) {
+	url := sysprobeclient.ModuleURL(sysconfig.NetworkTracerModule, "/debug/conntrack/cached")
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	return probeUtil.GetConnTrackCached()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(`GetConnTrackCached got non-success status code: url: %s, status_code: %d, response: "%s"`, req.URL, resp.StatusCode, data)
+	}
+
+	return data, nil
 }
 
 // GetSystemProbeConntrackHost queries conntrack/host, which uses netlink to return the list of NAT'd connections
-func GetSystemProbeConntrackHost(socketPath string) ([]byte, error) {
-	probeUtil, err := net.GetRemoteSystemProbeUtil(socketPath)
+func GetSystemProbeConntrackHost(client *http.Client) ([]byte, error) {
+	url := sysprobeclient.ModuleURL(sysconfig.NetworkTracerModule, "/debug/conntrack/host")
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	return probeUtil.GetConnTrackHost()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(`GetConnTrackHost got non-success status code: url: %s, status_code: %d, response: "%s"`, req.URL, resp.StatusCode, data)
+	}
+
+	return data, nil
 }
 
 // GetSystemProbeBTFLoaderInfo queries ebpf_btf_loader_info which gets where the BTF data came from
-func GetSystemProbeBTFLoaderInfo(socketPath string) ([]byte, error) {
-	probeUtil, err := net.GetRemoteSystemProbeUtil(socketPath)
+func GetSystemProbeBTFLoaderInfo(client *http.Client) ([]byte, error) {
+	url := sysprobeclient.DebugURL("/ebpf_btf_loader_info")
+	req, err := http.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
 		return nil, err
 	}
-	return probeUtil.GetBTFLoaderInfo()
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(`GetEbpfBtfInfo got non-success status code: url: %s, status_code: %d, response: "%s"`, req.URL, resp.StatusCode, data)
+	}
+
+	return data, nil
```

Comment:
nit: I think we could maybe have a generic method to get the response bytes from an URL, to avoid repeating this part.